presto-hive/src/test/java/com/facebook/presto/hive/TestOrcPageSourceMemoryTracking.java

```diff
@@ -0,0 +1,524 @@
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_LIB;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory.getStandardStructObjectInspector;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaStringObjectInspector;
+import static org.apache.hadoop.mapreduce.lib.output.FileOutputFormat.COMPRESS_CODEC;
+import static org.apache.hadoop.mapreduce.lib.output.FileOutputFormat.COMPRESS_TYPE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+public class TestOrcPageSourceMemoryTracking
+{
+    private static final TypeManager TYPE_MANAGER = new TypeRegistry();
+
+    private final Random random = new Random();
+    private final List<TestColumn> testColumns = ImmutableList.<TestColumn>builder()
+             .add(new TestColumn("p_empty_string", javaStringObjectInspector, () -> "", true))
+             .add(new TestColumn("p_string", javaStringObjectInspector, () -> Long.toHexString(random.nextLong()), false))
+             .build();
+    private final int numRows = 50000;
+
+    private File tempFile;
+    private TestPreparer testPreparer;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        tempFile = File.createTempFile("presto_test_topsfmt", "orc");
+        tempFile.delete();
+        testPreparer = new TestPreparer(tempFile.getAbsolutePath());
+    }
+
+    @AfterClass
+    public void tearDown()
+            throws Exception
+    {
+        tempFile.delete();
+    }
+
+    @Test
+    public void testPageSource()
+            throws Exception
+    {
+        ConnectorPageSource pageSource = testPreparer.newPageSource();
+
+        assertEquals(pageSource.getSystemMemoryUsage(), 0);
+
+        //*/
+        long memoryUsage = -1;
+        for (int i = 0; i < 20; i++) {
+            assertFalse(pageSource.isFinished());
+            Page page = pageSource.getNextPage();
+            assertNotNull(page);
+            Block block = page.getBlock(1);
+            if (memoryUsage == -1) {
+                assertBetweenInclusive(pageSource.getSystemMemoryUsage(), 180000L, 189999L); // Memory usage before lazy-loading the block
+                VARCHAR.getSlice(block, block.getPositionCount() - 1); // trigger loading for lazy block
+                memoryUsage = pageSource.getSystemMemoryUsage();
+            }
+            else {
+                assertEquals(pageSource.getSystemMemoryUsage(), memoryUsage);
+                VARCHAR.getSlice(block, block.getPositionCount() - 1); // trigger loading for lazy block
+                assertEquals(pageSource.getSystemMemoryUsage(), memoryUsage);
+            }
+        }
+        assertBetweenInclusive(memoryUsage, 460000L, 469999L); // Memory usage after lazy-loading the actual block
```

Comment:
Move this up to where you assign `memoryUsage`. It's hard to follow what this assertion is doing, and you only modify it in one place